internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -280,54 +283,67 @@ func createPatchGenerator(runtimeClient runtimeclient.Client, patch *clusterv1.C
 // applyPatchesToRequest updates the templates of a GeneratePatchesRequest by applying the patches
 // of a GeneratePatchesResponse.
 func applyPatchesToRequest(ctx context.Context, req *runtimehooksv1.GeneratePatchesRequest, resp *runtimehooksv1.GeneratePatchesResponse) error {
-	log := tlog.LoggerFrom(ctx)
-
 	for _, patch := range resp.Items {
-		log := log.WithValues("uid", patch.UID)
+		if err := applyPatchToRequest(ctx, req, patch); err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
-		// Get the request item the patch belongs to.
-		requestItem := getRequestItemByUID(req, patch.UID)
+func applyPatchToRequest(ctx context.Context, req *runtimehooksv1.GeneratePatchesRequest, patch runtimehooksv1.GeneratePatchesResponseItem) (reterr error) {
+	log := tlog.LoggerFrom(ctx).WithValues("uid", patch.UID)
 
-		// If a patch doesn't have a corresponding request item, the patch is invalid.
-		if requestItem == nil {
-			log.Infof("Unable to find corresponding request item with uid %q for the patch", patch.UID)
-			return errors.Errorf("unable to find corresponding request item for patch")
+	defer func() {
+		if r := recover(); r != nil {
+			log.Infof("Observed a panic when applying patch: %v\n%s", r, string(debug.Stack()))
+			reterr = kerrors.NewAggregate([]error{reterr, fmt.Errorf("observed a panic when applying patch: %v", r)})
```

Comment:
Could you include the patch name in the output here? It seems useful to users to have it on the condition.

I'm currently getting:


```
      message: 'error computing the desired state of the Cluster topology: failed
        to apply patches: observed a panic when applying patch: runtime error: invalid
        memory address or nil pointer dereference'

```

Reply:
If you tell me where I can get the patch name from, yes sure :) 

The best I can do afaik is the uid. But this will lead to infinite reconcile

Reply:
Could the error be wrapped with the patch name in l105?

Reply:
Nice one. Yup!

I was looking for the same when I implemented this PR, but I didn't think about looking up there.

Now wrapping the error in l.105